bokeh/charts/_builder.py

```diff
@@ -76,8 +81,26 @@ class Builder(HasProps):
 
     x_range = Instance(Range)
     y_range = Instance(Range)
+    y_names = Seq(String)
+    x_names = Seq(String)
```

Comment:
The idea was to support the pattern for all charts. Actually HeatMap is not considering it "the right way" but it should. I didn't spend much time on this for now because I think it's easier on the following PR to replace DataAdapter with DataFrame.

Do you have charts in mind that would not support this pattern?